content/post/file-reading-benchmarking/index.md

```diff
@@ -1,376 +1,31 @@
- *
- *  @param[in]   buff    16 bits buffer read from the DAT file.
- *  @param[out]  ts      12 entries array of 64 bits timestamps.
- *  @param[out]  x       12 entries array of 16 bits x addresses.
- *  @param[out]  y       12 entries array of 16 bits y addresses.
- *  @param[out]  p       12 entries array of 8 bit polarities.
- *
- *  @return      isEvent True when an event has been decoded.
- */
-bool decode_event(
-    const uint16_t buff, 
-    int64_t ts[12], 
-    int16_t x[12], 
-    int16_t y[12], 
-    uint8_t p[12]
-    ) {
-    const uint16_t mask_12b = 0xFFF; 
-    const uint16_t mask_11b = 0x7FF; 
-
-    static uint64_t tsHigh, tsLow = 0;
-    static int16_t baseX = 0; // Base x address for vectorized events.
-    static int16_t baseY = 0; // To remember the y value across events.
-    static int8_t baseP = 0; // To remember the p value across events.
-
-    int16_t numVectEvts = 0; 
-
-    bool isEvent = false; 
-
-    /** Resetting the polarity array. Our policy is that polarity can be 
-     *  either 0 or 1; by setting the elemnts in the polarity vector to 
-     *  a number larger than 1, we tell the user that those are non valid 
-     *  events.
-     */
-    int16_t i=0; 
-    for (i=0; i<12; i++)
-        p[i] = 99; 
-    
-    uint8_t evt_type = buff >> 12; 
-
-    switch (evt_type) {
-        case 0x0: // EVT_ADDR_Y.
-            baseY = buff & mask_11b;  
-            break; 
-        
-        case 0x2: // EVT_ADDR_X.
-            ts[0] = (tsHigh << 12) | tsLow; 
-            x[0] = buff & mask_11b; 
-            y[0] = baseY; 
-            p[0] = (buff >> 11) & 1; 
-            isEvent = true; 
-            break; 
-
-        case 0x3: // EVT_BASE_X.
-            baseX = buff & mask_11b; 
-            baseP = (buff >> 11) & 1; 
-            break; 
-
-        case 0x4: // VECT_12.
-            numVectEvts = 12; 
-        case 0x5: // VECT_8; 
-            if (numVectEvts != 12)
-                numVectEvts =  8;
-            int16_t mask = buff & mask_12b; 
-            for (i=0; i<numVectEvts; i++) {
-                if (mask & 1){
-                    ts[i] = (tsHigh << 6) | tsLow; 
-                    x[i] = baseX + i; 
-                    y[i] = baseY; 
-                    p[i] = baseP; 
-                    isEvent = true; 
-                }
-                mask = mask >> 1; 
-            }
-            break; 
-
-        case 0x6: // TIME_LOW.
-            tsLow = buff & mask_12b; 
-            break; 
-        
-        case 0x8: // TIME_HIGH.
-            tsHigh = buff & mask_12b; 
-    }
-
-    return isEvent; 
-}
-```
-
-This code is just for demonstration purposes, it won't actually work, since we need to take care of other things such as timestamp overflows. A working version of this code is provided [here](https://github.com/open-neuromorphic/expelliarmus/blob/cc9fbf1f53bfccd75c920e37d4ed94aa5aec3b1b/expelliarmus/src/evt3.c#L256).
-
-## Authors
```

Comment:
Shouldn't we list ourselves as authors?

Reply:
yes good point

Reply:
You can uncomment my part and add your description

Reply:
I added back your section!